src/panel-triggers/components/Problems/Problems.tsx

```diff
@@ -55,368 +63,401 @@ export const ProblemList = (props: ProblemListProps) => {
-        delete column.show;
-        result.push(column);
-      }
-    }
-    return result;
-  }, [panelOptions, handleTagClick]);
+  const handlePageSizeChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const newPageSize = Number(e.target.value);
+    reportPageSizeChange(newPageSize);
+    table.setPageSize(newPageSize);
+    onPageSizeChange?.(newPageSize, table.getState().pagination.pageIndex);
+  };
 
-  const pageSizeOptions = useMemo(() => {
+  // Calculate page size options
+  const pageSizeOptions = React.useMemo(() => {
     let options = [5, 10, 20, 25, 50, 100];
     if (pageSize) {
       options.push(pageSize);
-      options = _.uniq(_.sortBy(options));
+      options = Array.from(new Set(options)).sort((a, b) => a - b);
     }
     return options;
   }, [pageSize]);
 
   return (
-    <div className={cx('panel-problems', { [`font-size--${fontSize}`]: !!fontSize })} ref={rootRef}>
-      <ReactTable
-        data={problems}
-        columns={columns}
-        defaultPageSize={10}
-        pageSize={effectivePageSize}
-        pageSizeOptions={pageSizeOptions}
-        resized={panelOptions.resizedColumns}
-        minRows={0}
-        loading={loading}
-        noDataText="No problems found"
-        SubComponent={(props) => (
-          <ProblemDetails
-            {...props}
-            rootWidth={rootRef?.current?.clientWidth || 0}
-            timeRange={timeRange}
-            showTimeline={panelOptions.problemTimeline}
-            allowDangerousHTML={panelOptions.allowDangerousHTML}
-            panelId={panelId}
-            getProblemEvents={getProblemEvents}
-            getProblemAlerts={getProblemAlerts}
-            getScripts={getScripts}
-            onProblemAck={handleProblemAck}
-            onExecuteScript={onExecuteScript}
-            onTagClick={handleTagClick}
-            subRows={false}
-          />
+    <div className="react-table-v8-container">
+      <div className={`react-table-v8-wrapper ${loading ? 'is-loading' : ''}`}>
+        {loading && (
+          <div className="-loading -active">
+            <div className="-loading-inner">Loading...</div>
+          </div>
         )}
-        expanded={getExpandedPage(page)}
-        onExpandedChange={handleExpandedChange}
-        onPageChange={(newPage) => {
-          reportInteraction('grafana_zabbix_panel_page_change', {
-            action: newPage > page ? 'next' : 'prev',
-          });
-
-          setPage(newPage);
-        }}
-        onPageSizeChange={handlePageSizeChange}
-        onResizedChange={handleResizedChange}
-      />
-    </div>
-  );
-};
-
-interface HostCellProps {
-  name: string;
-  maintenance: boolean;
-}
-
-const HostCell: React.FC<HostCellProps> = ({ name, maintenance }) => {
-  return (
-    <div>
-      <span style={{ paddingRight: '0.4rem' }}>{name}</span>
-      {maintenance && <FAIcon customClass="fired" icon="wrench" />}
+        <table className="react-table-v8">
```

Comment:
react-table is using native elements: https://tanstack.com/table/latest/docs/framework/react/examples/basic